lib/network.dart

```
@@ -159,27 +166,29 @@ class NetworkImageWithRetry extends ImageProvider<NetworkImageWithRetry> {
       }
     }
 
-    if (instructions.alternativeImage != null)
-      return instructions.alternativeImage;
+    if (instructions.alternativeImage != null) {
+      return instructions.alternativeImage!;
+    }
 
     assert(lastFailure != null);
 
-    FlutterError.onError(FlutterErrorDetails(
-      exception: lastFailure,
+    FlutterError.onError!(FlutterErrorDetails(
```

The docs specifically say this can be set to null to ignore errors; this is not safe.

---

Sure, I can not disagree. However as tried my best to kept the same behaviour, looks like before there weren't any checks for null either, which made me think that at this point of the code there's _implicitly_ a onError defined.

---

This code has no way of guaranteeing that; it was simply not correct before in that edge case. Migration to null-safety should fix issues that are exposed by the migration, not explicitly codify that.

---

Would change it for `FlutterError.onError?` be enough?

---

Yes, if it's null'd then doing nothing for this line seems correct.